overlord/snapstate/snapstate.go

```diff
@@ -1273,6 +1276,16 @@ func doUpdate(ctx context.Context, st *state.State, names []string, updates []*s
 		updated = append(updated, name)
 	}
 
+	// Report a useful error if nothing can be updated and we have at
+	// least one auto-refresh change in progress.
+	if refreshAll && len(tasksets) == 0 {
+		for _, err := range updateErrs {
+			if _, ok := err.(*ChangeConflictError); ok {
+				return nil, nil, fmt.Errorf("auto-refresh has conflicting changes in progress")
```

Comment:
Would it make sense to include ChangeKind of the first ChangeConflictError? Otherwise looks fine to me.

Reply:
I like this suggestion too - should we put "changeKind: refresh" or "refresh-all" here? WDYT?

Reply:
ChangeKind is the kind of the operation already in progress, not the what is attempted

Reply:
Sorry, what I meant was including the ChangeKind of the operation already in progress in the error message (to give more context about what caused the conflict).